lib/generators/engine_cart/install_generator.rb

```diff
@@ -52,7 +52,12 @@ def add_gemfile_include
   # the below comes from engine_cart, a gem used to test this Rails engine gem in the context of a Rails app
   file = File.expand_path("Gemfile", ENV['ENGINE_CART_DESTINATION'] || ENV['RAILS_ROOT'] || File.expand_path("../spec/internal", __FILE__))
   if File.exist?(file)
-    eval_gemfile file
+    begin
+      eval_gemfile file
+    rescue Bundler::GemfileError => e
+       Bundler.ui.warn '[EngineCart] Skipping Rails application dependencies:'
+       Bundler.ui.warn e.message
+    end
```

Comment:
Yes, you get these errors immediately [1]. As long as the engine cart block is last in the Gemfile (and, in the context of a gem, I don't know what else you'd want to add there...).

I was able to reproduce it with stupid cases like defining a new restriction on the `rails` gem in my Gemfile. That's partly why I'd like to understand what changes people are actually making to run into this class of problem.

[1] https://github.com/bundler/bundler/blob/master/lib/bundler/dsl.rb#L96